manifests/keypair.pp

```diff
@@ -0,0 +1,41 @@
+define certs::keypair (
+  $key_pair,
+  $key_file,
+  $cert_file,
+  $manage_key  = false,
+  $key_owner   = undef,
+  $key_group   = undef,
+  $key_mode    = undef,
+  $manage_cert = false,
```

Comment:
Do we want to manage by default? Seems like we should default to manage the certificates and keys since we generate them and let users of the module opt out where necessary. In some cases, we weren't always "managing" the cert file but I think this was largely due to inexperience with puppet and only "managing" it when we needed to set user/mode specifically. Whereas, we probably ought to have been managing it the whole time to ensure puppet recreates the files when the original file changes or we run cert updates.